renku/core/commands/dataset.py

```diff
@@ -276,6 +276,10 @@ def add_to_dataset(
                     progress=progress,
                 )
 
+            if messages:
+                for msg in messages:
+                    click.echo(INFO + msg)
```

Comment:
I know we had previous similar implementation and this is just addition on top of it - with that said I think we should not do `click.echo` inside core commands since it could lead to leaking of messages to logs when used in service (or any other SDK context). I would propose we simply return those messages and then do click output inside of the CLI (since core module should not strictly depend on click - if we want to use it as SDK).

Reply:
Click was mostly removed in dataset.py after https://github.com/SwissDataScienceCenter/renku-python/pull/738 but has been slowly creeping back in. There's also https://github.com/SwissDataScienceCenter/renku-python/issues/737
So not sure if this PR is the right place to address this.

Reply:
https://github.com/SwissDataScienceCenter/renku-python/pull/1225 would this do?